prov/util/src/util_mem_monitor.c

```diff
@@ -81,9 +81,9 @@ void ofi_monitors_init(void)
 	uffd_monitor->init(uffd_monitor);
 	memhooks_monitor->init(memhooks_monitor);
 
-#if defined(HAVE_ELF_H) && defined(HAVE_SYS_AUXV_H)
+#if defined(HAVE_ELF_H) && defined(HAVE_SYS_AUXV_H) && ENABLE_MEMHOOKS_MONITOR
```

Comment:
Can we simplify this to ENABLE_MEMHOOKS_MONITOR only?  The other checks should be rolled into that

Reply:
Did you have an idea in mind? Like a define at the top of the file, or one generated by autoconf?

Reply:
I was thinking of one generated by autoconf.

Reply:
I've been looking down this path. uffd isn't hard, but memhooks is a little complicated. It's doable, but perhaps a little ugly. I'm still trying to figure out a solution for this. 

Proposal is this: HAVE_MEMHOOKS_MONITOR, HAVE_UFFD_MONITOR

What the autoconf for this looks like is TBD. Do you happen to have a @jsquyres on hand?

Reply:
It's a little tricky to test for `HAVE_FOO` kinds of things in `configure.ac` -- the contents of those tests are generally black boxes; all you get out is the `AC_DEFINE` (which puts something in `config.h`, but not necessarily a corresponding public shell variable that you can use later in `configure.ac`).

The typical way to do this is to put a compound check inside `config.h`.  Perhaps something like this:

```sh
# In configure.ac
AH_BOTTOM([
#if defined(HAVE_ELF_H) && defined(HAVE_SYS_AUXV_H) && ENABLE_MEMHOOKS
#define WE_WANTS_THE_MEMHOOKS 1
#else
#define WE_WANTS_THE_MEMHOOKS 0
#endif
])
```

Where `AH_BOTTOM` will put stuff at the bottom of the generated `config.h` for you.  Fun, right?  🎉 

```c
// In util_mem_monitor.c
#if WE_WANTS_THE_MEMHOOKS
    // .. do stuff with the memhooks ..
#endif
```

Reply:
Aha! See. Better than what I was going to do... 

```
dnl Check support to intercept syscalls
AC_CHECK_HEADERS([elf.h sys/auxv.h], [can_intercept=1], [can_intercept=0])

...
dnl enable compilation of memhooks provider if conditions satisfied
AS_IF([test $can_intercept -eq 1 -a $enable_memhooks -eq 1],
      [have_memhooks_monitor=1], [have_memhooks_monitor=0])

AC_DEFINE_UNQUOTED(HAVE_MEMHOOKS_MONITOR, [$have_memhooks_monitor],
        [Define to 1 if memhooks memory monitor is available])
```

Thanks @jsquyres . I think I'll take your approach.

Reply:
Maybe try converting AC_CHECK_HEADERS_ONCE(elf.h sys/auxv.h) to:
```
AC_CHECK_HEADERS([elf.h sys/auxv.h],
    [if disable option has not been set define HAVE_MEMHOOKS_MONITOR],
    ...)
```

Reply:
@jswaro - Your proposal isn't too far off from how HAVE_UFFD_UNMAP is defined.  With uffd, it uses a single have_uffd variable that carries through the checks, rather than using two.  Just need to pick the order that the checks are made.

Reply:
The two approaches are probably toe-MAY-toe vs. toe-MAH-toe.  I.e., both work; it's now a matter of style as to which one you prefer.